File: examples/RocketTestObjc/RocketTestObjc/ViewController.m
//
//  ViewController.m
//  RocketTestObjc
//
//  Created by Nick Bolton on 5/7/17.
//  Copyright © 2017 Pixelbleed LLC. All rights reserved.
//

#import "ViewController.h"
#import <RocketKit/RocketKit-Swift.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    RocketComponent *component = [[LayoutProvider shared] componentByIdentifier:@"9AF4A96E-8872-4C81-B2F9-5522214FD9D7"];
    if (component != nil) {
        component.textDescriptor.text = @"What in the world is happening here??";
        
        NSTimeInterval delayInSeconds = 3.0f;
        dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, delayInSeconds * NSEC_PER_SEC);
        dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
            component.textDescriptor.textAttributes.textAlignment = .center
            component.textDescriptor.text = "Weeee\nYo!"
            
            if var view = LayoutProvider.shared.view(with: component.identifier) {
                view.component = component
                view.updateText(animationDuration: 0.0)
            }
        });
    }
}

@end

File: examples/RocketTestSwift/RocketTestSwift/ViewController.swift
//
//  ViewController.swift
//  RocketTestSwift
//
//  Created by Nick Bolton on 5/7/17.
//  Copyright © 2017 Pixelbleed LLC. All rights reserved.
//

import UIKit
import RocketKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        let vc = RocketLayoutProvider.shared.buildViewController(withIdentifier: "542EA130-F283-494A-92DD-1BD57C8F072C")
        addChildViewController(vc)
        vc.view.frame = view.bounds
        view.addSubview(vc.view)
        vc.didMove(toParentViewController: self)
    }
}

